src/model.rs

```diff
@@ -159,14 +137,16 @@ impl ProofData {
 }
 #[derive(Serialize, Deserialize, Default, PartialEq, Eq, Clone, Encode, Decode, Hash)]
 pub struct ProofTransaction {
-    pub blobs_references: Vec<BlobReference>,
+    // TODO: investigate if we can remove blob_tx_hash. It can be reconstrustruced from HyleOutput attributes (blob + identity)
+    pub blob_tx_hash: TxHash,
```

Comment:
Je me dit que du coup il est impossible d'avoir plusieurs fois le même contract_name dans une blob_tx, sinon comment on sait pour quel blob la Proof est faite ?

est-ce qu'on ferait pas mieux d'avoir
```suggestion
    pub blobs_references: BlobReference,
```
?

Reply:
C'est effectivement pas très explicite dans la PR :sorry:
--> En faisant le réfacto, j'ai réalisé qu'il y avait qlq champs qui n'était pas nécessaire et qu'on pouvait faire sauter. Notamment le BlobIndex !! En effet, quand on recoit une transaction, on la vérifie et on stock HyleOutput... qui contient l'info de l'index !!

Reply:
On utilise donc l'info venant de HyleOutput lorsque l'on save l'info pour la BlobTx unsettled: 
https://github.com/Hyle-org/hyle/pull/338/files#diff-9ab274ad93a09b4c4a52efbf2c8f54279c3ba15b3feb5653470ead5f4e3727daR81

Reply:
Et pour pousser la logique jusqu'au bout, on a en fait pas besoin de blob_tx_hash, puisqu'on est sensé pouvoir le reconstruire à partir de HyleOutput...
Mais ça faisait refaire des hash un peu à gogo, donc j'ai mis un TODO pour le moment

Reply:
oui ok je vois, on donne l'index au moment de la génération de la preuve...

mais de même pour le tx_hash du coup, on est censé le donner à la génération de la preuve et la retourner dans l'output... à faire dans une PR séparée oui

Reply:
après, pour le TxHash, si on peut éviter de le passer à la génération de la preuve et l'avoir dans la ProofTx, ça fait ça de moins de donnée qui passe dans le prover, et donc moins coûteux